core/src/driver/scheduler/evm.rs

```diff
@@ -28,26 +31,34 @@ impl EvmScheduler {
         driver: Arc<dyn StableXDriver>,
         config: AuctionTimingConfiguration,
     ) -> Self {
-        EvmScheduler {
+        Self {
             driver,
             exchange,
+            sleep: Box::new(AsyncSleep),
             config,
             last_batch: None,
         }
     }
 
     /// Creates a new scheduler with the default configuration.
     #[cfg(test)]
-    pub fn with_defaults(
+    pub fn with_defaults_and_sleep(
         exchange: Arc<dyn StableXContract>,
         driver: Arc<dyn StableXDriver>,
+        sleep: Box<dyn AsyncSleeping>,
     ) -> Self {
-        EvmScheduler::new(exchange, driver, AuctionTimingConfiguration::default())
+        Self {
+            exchange,
+            driver,
+            sleep,
+            config: AuctionTimingConfiguration::default(),
+            last_batch: None,
+        }
     }
 
     /// Runs the scheduler for a single iteration.
-    fn step(&mut self) -> Result<()> {
-        let batch_id = self.current_solving_batch()?;
+    async fn step(&mut self) -> Result<()> {
```

Comment:
One other alternative is to make `step() -> Result<Option<Solution>>` and then move the sleep into the `Scheduler::start` loop. This would effectively split `step` into a `run` and `submit` methods. Also if we are going to use `AsyncSleeping` then maybe we should have `step` just wait for the next batch and wait to submit instead of sleeping in two different places with two different methods (`thread::sleep` and `AsyncSleeping::sleep`).

Reply:
I iike the proposal of splitting the functions and having the sleep in one place. This makes it a bit closer to a state machine while presumably makes the factoring simpler as we break the large step function into two.